pygfx/utils/show.py

```diff
@@ -11,56 +14,104 @@
     Scene,
     PerspectiveCamera,
     OrbitController,
-    Light,
     AmbientLight,
     DirectionalLight,
 )
 
 
-def show(object: WorldObject, up=None):
-    """Visualize a given WorldObject in a new window with an interactive camera.
+def show(
+    object: WorldObject,
+    up=None,
+    *,
+    canvas=None,
+    renderer=None,
+    controller=None,
+    camera=None,
+    before_render=None,
+    after_render=None,
+    draw_function=None
+):
+    """Display a WorldObject
+
+    This function provides you with the basic scaffolding to visualize a given
+    WorldObject in a new window. While it does add scaffolding, it aims to be
+    fully customizable so that you can replace each piece as needed.
+
+    Parameters
+    ----------
+        object : gfx.WorldObject
+            The object to show.
+        up : gfx.Vector3
+            If set, set the default camera controller's up vector to this value.
+        canvas : WgpuCanvas
+            The canvas to use to display the object.
+        renderer : gfx.Renderer
+            The renderer to use while drawing the scene. If set, `canvas` is
+            ignored.
+        controller : gfx.Controller
+            The camera controller to use.
+        camera : gfx.Camera
+            The camera to use.
+        before_render : Callable
+            A callback that will be executed during each draw call before a new
+            render is made.
+        after_render : Callable
+            A callback that will be executed during each draw call after a new
+            render is made.
+        draw_function : Callable
+            Replaces the draw callback with a custom one. If set both
+            `before_render` and `after_render` will have no effect.
 
-    Parameters:
-        object (WorldObject): The object to show.
-        up (Vector3): Optional. Configure the up vector for the camera controller.
     """
-    from wgpu.gui.auto import WgpuCanvas, run
 
     if isinstance(object, Scene):
         scene = object
     else:
         scene = Scene()
         scene.add(object)
 
-        background = Background(None, BackgroundMaterial((0, 1, 0, 1), (0, 1, 1, 1)))
+        dark_gray = np.array((169, 167, 168, 255)) / 255
+        light_gray = np.array((100, 100, 100, 255)) / 255
+
+        background = Background(None, BackgroundMaterial(light_gray, dark_gray))
         scene.add(background)
+        scene.add(AmbientLight(0.2))
+
+    if not camera:
+        camera = PerspectiveCamera(70, 16 / 9)
+        camera.add(DirectionalLight(0.8))
+        scene.add(camera)
```

Comment:
```suggestion
        if scene is not object:
            camera.add(DirectionalLight(0.8))
            scene.add(camera)
```

Reply:
How would this work for situations where we pass a scene to the renderer but don't pass a camera?

Reply:
Then we create a camera, but don't add it to the scene. The camera does not have to be in the scene for it to work. The reason it is added to the scene in these lines, is to make the light source move along with it (as if you had a head-light).

Reply:
I wasn't quite sure how to address this comment, especially after the reactor. Since I'm not sure what we were after here, I can't really say if this is resolved now or still something to improve. @almarklein could you have a look and let me know if we can resolve this?